From 18b61f18d052d3138a42057bcfbe8b86d0d681e2 Mon Sep 17 00:00:00 2001
From: Shivani Raval <sraval@habana.ai>
Date: Tue, 29 Apr 2025 10:27:28 +0300
Subject: [PATCH] [SW-227530] gaudi3: rate limit all intr prints

Currently storm of prints of some interrupts,
So using LOG_RL_INF API to rate limit the prints.
Now prints will be max 6 within 1sec time frame.

Change-Id: I353b7533945bdcffe10ea23a94f8ae53d0990743
Signed-off-by: Shivani Raval <sraval@habana.ai>
---

diff --git a/src/event.c b/src/event.c
index 698b7f9..5a68b06 100644
--- a/src/event.c
+++ b/src/event.c
@@ -12,6 +12,7 @@
 
 #define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
 #include <zephyr/logging/log.h>
+#include <log_wrapper.h>
 LOG_MODULE_REGISTER(EVENT_LOG_ID);
 
 static struct hl_event_config evt_config;
@@ -22,6 +23,9 @@
 
 	LOG_INF("arc event booted");
 
+	/* Initialize the log pool memory for rate limiting logging feature */
+	common_log_pool_init();
+
 	ret = hl_event_prepare(&evt_config);
 	if (ret) {
 		LOG_ERR("event prepare failed err: %d", ret);
diff --git a/src/gaudi3/interrupt.c b/src/gaudi3/interrupt.c
index ae858cc..b551ee8 100644
--- a/src/gaudi3/interrupt.c
+++ b/src/gaudi3/interrupt.c
@@ -124,6 +124,7 @@
 #include <pif_regs.h>
 #include "ras.h"
 #include "rma.h"
+#include <log_wrapper.h>
 
 #define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
 #include <zephyr/logging/log.h>
@@ -4563,23 +4564,17 @@
 		}
 	}
 
-	/*
-	 * TODO: This is a workaround to avoid mc intr storms.
-	 * Fix this with console log re-design
-	 */
-	if (!(binned_intr_map.comp_type == INT_COMP_TYPE_MC &&
-		binned_intr_map.grp_type == INT_GRP_TYPE_SPI)) {
-		LOG_INF("%s[%s, DIE%d, %s, INSTANCE%d, %s] [ARC_IDX: %d AGG_IDX: %d] [%s: %d]",
-			(config->state == HL_EVENT_INIT)?"[EARLY INTR] ":"",
-			gaudi3_get_interrupt_comp_name(binned_intr_map.comp_type),
-			binned_intr_map.die_id,
-			gaudi3_get_interrupt_hdcore_name(binned_intr_map.hdcore_type),
-			binned_intr_map.comp_inst,
-			gaudi3_get_interrupt_grp_name(binned_intr_map.grp_type),
-			binned_intr_map.cpu_id, binned_intr_map.index,
-			"NOTIFY_LKD",
-			binned_intr_map.notify_lkd);
-	}
+	LOG_RL_INF(intr_info,
+		"%s[%s, DIE%d, %s, INSTANCE%d, %s] [ARC_IDX: %d AGG_IDX: %d] [%s: %d]",
+		(config->state == HL_EVENT_INIT)?"[EARLY INTR] ":"",
+		gaudi3_get_interrupt_comp_name(binned_intr_map.comp_type),
+		binned_intr_map.die_id,
+		gaudi3_get_interrupt_hdcore_name(binned_intr_map.hdcore_type),
+		binned_intr_map.comp_inst,
+		gaudi3_get_interrupt_grp_name(binned_intr_map.grp_type),
+		binned_intr_map.cpu_id, binned_intr_map.index,
+		"NOTIFY_LKD",
+		binned_intr_map.notify_lkd);
 
 	if (config->state == HL_EVENT_INIT && !process_early_irq) {
 		pdata = k_malloc(sizeof(struct pend_fifo_item_t));
